### PR TITLE
Adds `denom` argument to count `s_*` functions

### LINKED ISSUE #1325: Add `denom` parameter to `s_count_occurrences_by_grade()`
Add `denom` parameter to `s_count_occurrences_by_grade()` so that it can be used in `count_occurrences_by_grade()`.

### PR DESCRIPTION
**Pull request by edelarua**
# Pull Request

Fixes #1325

Adds `denom` to `s_count_occurrences_by_grade()` as well as `s_count_cumulative()` and `s_count_missed_doses()`.

Downstream changes:

- [ ] https://github.com/insightsengineering/scda.test/pull/155
- [ ] https://github.com/insightsengineering/tlg-catalog/pull/276

### CODE REVIEW

File: tests/testthat/_snaps/count_occurrences_by_grade.md [hunk 1]
```diff
@@ -400,3 +400,19 @@
...
+                      A           B    
+      —————————————————————————————————
+      LOW                              
+        MILD          0       1 (100%) 
+        MODERATE      0           0    
+        SEVERE     2 (100%)       0    
+      HIGH                             
+        MILD          0       1 (50.0%)
+        MODERATE   1 (100%)   1 (50.0%)
+        SEVERE        0           0
```
Comment (shajoezhu) [inline tests/testthat/_snaps/count_occurrences_by_grade.md:417]: This is perfect! @edelarua , I was wondering if you could add a summerizie row, so the total count is more obvious in this snapshot

**Comment by shajoezhu** [comment]
Hi @edelarua , this looks really great! Thank you. I was wondering if you could update downstream scda.test and tlg-c accordingly please. use staged dependencies.

@pawelru , I was wondering if we could set a meeting and follow-up on a poc, demo how to the above with pak please? thank you!

File: tests/testthat/_snaps/count_occurrences_by_grade.md (hunk 1 above)
Comment (edelarua) [reply tests/testthat/_snaps/count_occurrences_by_grade.md]: Snapshot updated

**Review by shajoezhu — APPROVED** [review]
this is good to go. Thanks @edelarua  lets wait for the pak discussion, and merge after

**Pull request merged**